eoscostarica/eos-local
# Apollos link state and Client

This a proposal to implement `apollo-link-state` in replacement of `react-redux`.

### Motivation
Apollo has removed redux from their library and they created their own library for local state management. Using Redux and Apollo at same time could be tricky and over the time `react-redux` won't be use much

### Changes
* Replace Apollo boost and install all the packages since ABoost will be deprecated soon and does not support some features such as `devTools` and `subscriptions`
* Create the configuration for Apollo client
* Propose GraphQL structure to handle local state and queries, resolvers, mutations
* Implement a example using `apollo-link-state` by binding the state with the greeting input

### Steps to test

- Download and open Apollo dev tools
- Go to Cache
- Start writing down on the input at the top of the page
- Greetings coming from the Postgres should be cached as well

### References

- https://github.com/apollographql/apollo-client/issues/3778
- https://www.apollographql.com/docs/react/advanced/boost-migration.html

closes #43
